compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.cpp

```diff
@@ -263,16 +263,28 @@ SmallVector<int64_t> getWorkgroupSize(IREE::HAL::ExecutableExportOp exportOp) {
   return {};
 }
 
+llvm::Optional<int64_t> getSubgroupSize(
+    IREE::HAL::ExecutableExportOp exportOp) {
+  if (IntegerAttr attr = exportOp.getSubgroupSizeAttr()) {
+    return attr.getValue().getSExtValue();
+  }
+  return {};
+}
+
 void setTranslationInfo(IREE::HAL::ExecutableExportOp exportOp,
                         IREE::Codegen::TranslationInfoAttr translationInfo,
-                        ArrayRef<int64_t> workgroupSize) {
+                        ArrayRef<int64_t> workgroupSize,
+                        Optional<int64_t> subgroupSize) {
```

Comment:
I am wondering if you even want to make it part of this. I did this, but I dont think it was a good idea to start with that `workgroupSize`  was part of this. I used it as a crutch, that might not be worth it. COuld we instead set the subgroup size at call site. Bonus points to moving setting `workgroupSize` also out of this method

Reply:
Would you mind to elaborate a bit more on `set the subgroup size at call site`? Do you mean having `setTranslationInfo` only setting `IREE::Codegen::TranslationInfoAttr`, and set workgroup/subgroup size separately on the export op?

Reply:
I guess that's what you meant. SG; let me move it to separate utility functions.

Reply:
https://github.com/iree-org/iree/pull/11400 to split translation info and workgroup size setting.

Reply:
Also split the subgroup setting out.